### PR TITLE
Add Git module to Faker

### PR DESCRIPTION
**Pull request by danielleadams**
Adds a `git` module to the library. Support functionality for https://github.com/Marak/faker.js/issues/659.

### Includes
* `.branch()` (ex. `pixel-compress`)
* `.commitSha()` (ex. `2regs2p9kypy8my7gm5ws1top4cnr4ws4jni3mic`)
* `.commitMessage()` (ex. `compress primary pixel`)
* `.shortSha()` (ex. `ppxmq7r`)
* `.commitEntry()` (ex:
```
commit oliatodogpm4ursl6yqmw1983sub8qgsyvybc5jv
Merge: 9vj3pt8 mnjggpg
Author: Joey Cronin <Gayle.Littel15@hotmail.com>
Date: Wed Jun 27 2018 05:44:27 GMT-0400 (EDT)

    hack virtual interface

```
)

### CODE REVIEW

**Comment by Marak** [comment]
@danielleadams -

After a quick review this looks good. Thanks!

I'm thinking it might make sense to create a composite method now for generating a commit entry similar to what would appear as if I ran `git log` locally. Feel free to add this to the PR or I can try to for the next release.

One thing to note for the future is that our `ReadMe` modifications are actually done automatically through the build process so in this case method names would be pulled from the in-line jsdocs.

**Comment by danielleadams** [comment]
Great! As for the commit entry.. something like this?

```
commit 3ae345c60d3a90f50925cfda8939971936b53998
Author: Danielle Adams <danielle.adams@me.com>
Date:   Tue Jun 26 16:04:22 2018 -0400

    add git module to lib
```

I think it makes sense to add this to the PR. I can also remove the `ReadMe` changes since I'll be adding another method 👌

**Comment by Marak** [comment]
@danielleadams -

Exactly yes.

I was also considering a random switch that would also occasionally ( maybe 20% ) add a `Merge` line after `commit` to make more realistic looking fake results.

For example:

```
commit d58ee47f162cea98c064d5200ca9acf422db3974
Merge: 23c05db 9d47fd0
Author: Marak <marak.squires@gmail.com>
Date:   Wed Sep 13 10:44:29 2017 -0400

    this is a commit message
```

Let me know if you can think up any additional ideas or API methods we should add for `git` module now that we are adding it. I think we've gotten the basics covered for now.

**Comment by danielleadams** [comment]
@Marak made the adjustments and added the commit entry functionality. One thing to note is the JS Date object was being a little stubborn, so it doesn't follow the _exact_ git date format... but it's pretty close.
git `Wed Jun 27 20:31:44 2018 -0400`
vs.
`new Date().toString()` or `faker.date.recent().toString()` `Wed Jun 27 2018 20:34:26 GMT-0400 (EDT)`

**Comment by Marak** [comment]
Awesome!

The date format should be OK. Someone can always fix it later if they care to.

Marking this PR for merge in next release.

**Comment by danielleadams** [comment]
Ok awesome! Thanks for reviewing this 😃

File: lib/git.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+   * @method faker.git.shortSha
+   */
+  self.shortSha = function() {
+    var shortSha = "";
+
+    for (var i = 0; i < 7; i++) {
+      shortSha += faker.random.alphaNumeric();
+    }
+
+    return shortSha;
```
Comment (uhoh-itsmaciek) [inline lib/git.js]: Not sure it matters but shas are hex, so just `[0-9a-f]`.

Comment (danielleadams) [reply lib/git.js]: good catch! I pushed up a change to reflect correct chars.

**Comment by Marak** [comment]
Merged into `master`

Thank you for your contribution @danielleadams

**Pull request closed without merging**